Code/Framework/AzCore/AzCore/Component/ComponentApplication.cpp

```diff
@@ -1219,129 +1221,81 @@ namespace AZ
         struct GemModuleLoadData
         {
-            AZ::OSString m_gemName;
+            AZ::OSString m_gemModuleName;
             AZStd::vector<AZ::OSString> m_dynamicLibraryPaths;
             bool m_autoLoad{ true };
         };
 
-        struct GemModuleVisitor
-            : AZ::SettingsRegistryInterface::Visitor
-        {
-            GemModuleVisitor()
-                : m_gemModuleKey{ AZ::SettingsRegistryInterface::FixedValueString::format("%s/Gems", AZ::SettingsRegistryMergeUtils::OrganizationRootKey) }
-            {}
 
-            AZ::SettingsRegistryInterface::VisitResponse Traverse(AZStd::string_view path, AZStd::string_view,
-                AZ::SettingsRegistryInterface::VisitAction action, AZ::SettingsRegistryInterface::Type) override
+        using FixedValueString = SettingsRegistryInterface::FixedValueString;
+        using Type = SettingsRegistryInterface::Type;
+        AZStd::vector<GemModuleLoadData> modulesLoadData;
+        auto GemModuleVisitor = [&settingsRegistry = *m_settingsRegistry, &modulesLoadData]
+        (AZStd::string_view gemRootObjectPath, AZStd::string_view, Type)
+        {
+            auto VisitGemObjectFields = [&settingsRegistry, &modulesLoadData](AZStd::string_view jsonPath,
+                AZStd::string_view fieldName, Type)
             {
-                if (action == AZ::SettingsRegistryInterface::VisitAction::Begin)
+                AZStd::string_view gemModuleName = fieldName;
+                auto FindGemModuleLoadEntry = [gemModuleName](const GemModuleLoadData& moduleLoadData)
                 {
-                    // Strip off the last JSON pointer key from the path and if it matches the gem module key then add an entry
-                    // to the ModulesLoadData array
-                    AZStd::optional<AZStd::string_view> gemNameKey = AZ::StringFunc::TokenizeLast(path, "/");
-                    if (path == m_gemModuleKey && gemNameKey && !gemNameKey->empty())
-                    {
-                        AZStd::string_view gemName = gemNameKey.value();
-                        auto FindGemModuleLoadEntry = [gemName](const GemModuleLoadData& moduleLoadData)
-                        {
-                            return gemName == moduleLoadData.m_gemName;
-                        };
-
-                        if (auto foundIt = AZStd::find_if(m_modulesLoadData.begin(), m_modulesLoadData.end(), FindGemModuleLoadEntry);
-                            foundIt == m_modulesLoadData.end())
-                        {
-                            m_modulesLoadData.emplace_back(GemModuleLoadData{ gemName });
-                        }
-                    }
-                }
+                    return gemModuleName == moduleLoadData.m_gemModuleName;
+                };
 
-                return AZ::SettingsRegistryInterface::VisitResponse::Continue;
-            }
+                auto foundIt = AZStd::ranges::find_if(modulesLoadData, FindGemModuleLoadEntry);
```

Comment:
When did `AZStd::ranges` get added!? 🤯 We should totally make more people aware this, that's really cool!

Reply:
We only have partial ranges support.

The current implementations are isolated in the https://github.com/o3de/o3de/tree/development/Code/Framework/AzCore/AzCore/std/ranges.
Many of the [ranges algorithms](http://eel.is/c++draft/algorithms) and some of the [ranges views](http://eel.is/c++draft/ranges) are not implemented.

We have implemented all the C++20 "[concepts](https://github.com/o3de/o3de/tree/development/Code/Framework/AzCore/AzCore/std/concepts)" though they have to be used using SFINAE.

Reply:
Interesting! Still worth a call out, if it simplifies call sites and makes reading code easier I'm all for encouraging using this and raising awareness